runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/Timer.java

```diff
@@ -40,14 +49,26 @@
 @AutoValue
 public abstract class Timer<T> {
 
-  /** Returns a timer for the given timestamp with a {@code null} payload. */
-  public static Timer<Void> of(Instant time) {
-    return of(time, (Void) null);
+  /**
+   * Returns a timer for the given timestamp with a user specified payload.
+   *
+   * @return
+   */
+  // TODO(BEAM-9562): Plumb through actual Timer fields.
+  public static <T> Timer<T> of(
```

Comment:
May I suggest you create these two static methods for creating the two common cases?
.of(userKey, dynamicTimerTag, fireTimestamp, holdTimestamp, windows, pane);
.cleared(userKey, dynamicTImerTag);

this way you can make sure that dynamicTimerTag, fireTimestamp, holdTimestamp, windows and pane are not null

Reply:
Thanks for you suggestion! I really like this idea. I planned to refine the signature when moving to data channel, but I can do it in this PR.